File: raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var sound = number.IsFactorOf(3, "Pling")
            + number.IsFactorOf(5, "Plang")
            + number.IsFactorOf(7, "Plong");

        if (string.IsNullOrEmpty(sound))
            return $"{number}";
        else
            return sound;
    }

    private static string IsFactorOf(this int number, int factor, string sound) =>
        number % factor == 0 ? sound : string.Empty;
}

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let private factors =
    [(3, "Pling"); (5, "Plang"); (7, "Plong")]

let private isFactorOf factor number =
    number % factor = 0

let private concatSounds soundsSoFar nextSound =
    sprintf "%s%s" soundsSoFar nextSound

let convert (number: int): string =
    let sounds = [
        for (factor, sound) in factors do
            if number |> isFactorOf factor
                then yield sound
    ]

    match sounds with
    | [] -> sprintf "%i" number
    | sounds -> ("", sounds) ||> List.fold concatSounds